homeassistant/components/evohome.py

```diff
@@ -0,0 +1,690 @@
+
+    One controller with 0+ heating zones (e.g. TRVs, relays) and, optionally, a
+    DHW controller.  Does not work for US-based systems.
+    """
+# 0. internal data, such as installation, state & timers...
+    hass.data[DATA_EVOHOME] = {}
+
+    domain_data = hass.data[DATA_EVOHOME]
+    domain_data['timers'] = {}
+
+# 1. pull the configuration parameters...
+    domain_data['params'] = dict(config[DOMAIN])
+    # scan_interval - rounded up to nearest 60 secs, with a minimum value
+    domain_data['params'][CONF_SCAN_INTERVAL] \
+        = (int((config[DOMAIN][CONF_SCAN_INTERVAL] - 1) / 60) + 1) * 60
+    domain_data['params'][CONF_SCAN_INTERVAL] = \
+        max(domain_data['params'][CONF_SCAN_INTERVAL], MIN_SCAN_INTERVAL)
+
+    if _LOGGER.isEnabledFor(logging.DEBUG):  # then redact username, password
+        tmp = dict(domain_data['params'])
+        tmp[CONF_USERNAME] = 'REDACTED'
+        tmp[CONF_PASSWORD] = 'REDACTED'
+
+        _LOGGER.debug("setup(): Configuration parameters: %s", tmp)
+
+# 2. instantiate the client
+    from evohomeclient2 import EvohomeClient
+
+    _LOGGER.debug("setup(): API call [4 request(s)]: client.__init__()...")
+
+    try:
+        # there's a bug in evohomeclient2 v0.2.7: the client.__init__() sets
+        # the root loglevel (debug=?), so must remember it now...
+        log_level = logging.getLogger().getEffectiveLevel()
+
+        client = EvohomeClient(
+            domain_data['params'][CONF_USERNAME],
+            domain_data['params'][CONF_PASSWORD],
+            debug=False
+        )
+        # ...then restore the it to what it was before instantiating the client
+        logging.getLogger().setLevel(log_level)
+
+    except requests.RequestException as err:
+        if str(HTTP_BAD_REQUEST) in str(err):
+            # this happens when bad user credentials are supplied
+            _LOGGER.error(
+                "Failed to establish a connection with evohome web servers, "
+                "Check your username (%s), and password are correct."
+                "Unable to continue. Resolve any errors and restart HA.",
+                domain_data['params'][CONF_USERNAME]
+            )
+        else:
+            # back off and try again later
+            raise PlatformNotReady(err)
+
+    finally:  # redact username, password as no longer needed
+        del domain_data['params'][CONF_USERNAME]  # = 'REDACTED'
+        del domain_data['params'][CONF_PASSWORD]  # = 'REDACTED'
+
+    domain_data['client'] = client
+
+
+# 3. REDACT any installation data we'll never need
+    if client.installation_info[0]['locationInfo']['locationId'] != 'REDACTED':
+        for loc in client.installation_info:
+            loc['locationInfo']['locationId'] = 'REDACTED'
+            loc['locationInfo']['streetAddress'] = 'REDACTED'
+            loc['locationInfo']['city'] = 'REDACTED'
+            loc['locationInfo']['locationOwner'] = 'REDACTED'
+            loc[GWS][0]['gatewayInfo'] = 'REDACTED'
+
+# 3a.
```

Comment:
These comments are getting out of hand. What does `3a` mean to anyone but you? Also, having all of these 0-indent comment lines makes your code really hard to read

Reply:
Removed.

Reply:
Removed all 0-indent comments, except those at the start of the code.